src/server/plugins/engine/pageControllers/FileUploadPageController.ts

```diff
@@ -131,7 +132,62 @@ export class FileUploadPageController extends QuestionPageController {
     }
   }
 
-  makePostRouteHandler() {
+  makeGetItemDeleteRouteHandler() {
+    return async (
+      request: FormRequest,
+      h: Pick<ResponseToolkit, 'redirect' | 'view'>
+    ) => {
+      const { path, viewModel } = this
+      const { params } = request
+
+      const state = await this.getState(request)
+      const uploadState = state.upload?.[path] ?? { files: [] }
+
+      const fileToRemove = uploadState.files.find(
+        (file) => file.uploadId === params.itemId
+      )
+
+      if (!fileToRemove) {
+        return Boom.notFound('File to delete not found')
```

Comment:
Would be great to place these error messages in an enum so we only have one place to change them

Reply:
Mind if we start a separate discussion on this?

Need a way to support template strings too in lots of our error messages, but sounds good still